src/Resources/ResourceManager/Implementation/Deployments/NewAzureDeploymentWhatIfCmdlet.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.Deployments
+{
+    using System;
+    using System.Management.Automation;
+    using Common.ArgumentCompleters;
+    using Management.ResourceManager.Models;
+    using SdkModels.Deployments;
+    using WindowsAzure.Commands.Utilities.Common;
+
+    /// <summary>
+    /// Creates a new deployment what-if.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New,
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DeploymentWhatIf", SupportsShouldProcess = true,
+         DefaultParameterSetName = SubscriptionParameterSetWithParameterlessTemplateFile),
+     OutputType(typeof(PSWhatIfOperationResult))]
+    public class NewAzureDeploymentWhatIfCmdlet : DeploymentWhatIfCmdletWithParameters, IDynamicParameters
```

Comment:
>NewAzureDeploymentWhatIfCmdlet [](start = 17, length = 30)

do we want to mark the cmdlet as preview?

Reply:
Yeah, we probably should.

Reply:
Just wanted to be sure: we cannot mark a single cmdlet as preview, can we?

Otherwise I have merged the latest changes so Prerelease of the module is set to preview:
https://github.com/Azure/azure-powershell/blob/bc8e967c87fdc9dfd68351cd322699049083946a/src/Resources/Resources/Az.Resources.psd1#L179

Reply:
I thought there's support for it. @isra-fel do you know if we can mark a cmdlet as preview?